services/user/Supervisor/ui/src/component-loading/loader.ts

```diff
@@ -83,6 +81,76 @@ async function getNonstandardWasiImports(): Promise<ImportDetails> {
     };
 }
 
+async function generateWasiShimCode(): Promise<string> {
+    // Import the JCO preview2-shim modules at build time and create a clean wrapper
+    try {
+        const preview2Shim = await import('@bytecodealliance/preview2-shim') as {
+            cli: unknown;
+            clocks: unknown;
+            filesystem: unknown;
+            io: unknown;
+            random: unknown;
+        };
+        
+        // Extract the specific interfaces we need
+        const { cli, clocks, filesystem, io, random } = preview2Shim;
+        
+        // Test that the objects are available
+        if (!cli || !clocks || !filesystem || !io || !random) {
+            throw new Error('Missing required shim modules');
+        }
+        
+        // Create a self-contained module that directly calls the JCO functions
+        // This avoids complex serialization by creating a functional wrapper
+        const shimCode = `
```

Comment:
I think this should be put in the shims directory as a file and imported with `?raw` (once) into a string.

Reply:
done